api/src/db/models/user_models.py

```diff
@@ -0,0 +1,39 @@
+import uuid
+
+from sqlalchemy import ForeignKey
+from sqlalchemy.dialects.postgresql import UUID
+from sqlalchemy.orm import Mapped, mapped_column, relationship
+
+from src.adapters.db.type_decorators.postgres_type_decorators import LookupColumn
+from src.db.models.base import ApiSchemaTable, TimestampMixin
+from src.db.models.lookup_models import LkExternalUserType
+
+
+class User(ApiSchemaTable, TimestampMixin):
+    __tablename__ = "user"
+
+    user_id: Mapped[uuid.UUID] = mapped_column(UUID, primary_key=True)
+
+
+class LinkExternalUser(ApiSchemaTable, TimestampMixin):
+    __tablename__ = "link_external_user"
+
+    link_external_user_id: Mapped[uuid.UUID] = mapped_column(UUID, primary_key=True)
+
+    external_user_id: Mapped[str]
```

Comment:
We should probably index/unique this since it'll be what we actually query the table by most of the time.

```suggestion
    external_user_id: Mapped[str] = mapped_column(index=True, unique=True)
```